core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -653,6 +619,39 @@ class Partition(val topicPartition: TopicPartition,
     }
   }
 
+  // Public visibility for tests
+  def updateAssignmentAndIsr(assignment: Seq[Int], isr: Set[Int]): Unit = {
+    val replicas = assignment
+      .filter(_ != localBrokerId)
+      .map(id => new Replica(id, topicPartition))
+    updateReplicasAndIsr(assignment, replicas, isr)
+  }
+
+  // Public visibility for tests
+  def updateReplicasAndIsr(assignment: Seq[Int], defaultRemoteReplicas: Seq[Replica], isr: Set[Int]): Unit = {
```

Comment:
This is kind of an awkward signature. It is forcing us to do some redundant validations when called in the context of `updateAssignmentAndIsr`. As far as I can tell, it has just one use in test cases. Possibly the test case can be refactored to use `updateAssignmentAndIsr` and `Partition.updateFollowerFetchState`?

Reply:
That is correct! We may be able to change the test as you described. Let me look into that.

Reply:
Done.